presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationSmokeTest.java

```diff
@@ -13,25 +13,106 @@
  */
 package io.prestosql.plugin.kafka;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
 import io.prestosql.plugin.kafka.util.TestingKafka;
+import io.prestosql.spi.connector.SchemaTableName;
+import io.prestosql.spi.type.BigintType;
+import io.prestosql.spi.type.Type;
 import io.prestosql.testing.AbstractTestIntegrationSmokeTest;
 import io.prestosql.testing.QueryRunner;
 import io.prestosql.tpch.TpchTable;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.ByteArraySerializer;
 import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.UUID;
+
+import static org.apache.kafka.clients.producer.ProducerConfig.ACKS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
 
 public class TestKafkaIntegrationSmokeTest
         extends AbstractTestIntegrationSmokeTest
 {
     private TestingKafka testingKafka;
+    private String topicName;
 
     @Override
     protected QueryRunner createQueryRunner()
             throws Exception
     {
         testingKafka = new TestingKafka();
-        return KafkaQueryRunner.builder(testingKafka)
+        topicName = "test_raw_" + UUID.randomUUID().toString().replaceAll("-", "_");
+
+        Map<SchemaTableName, KafkaTopicDescription> extraTopicDescriptions = ImmutableMap.<SchemaTableName, KafkaTopicDescription>builder()
+                .put(new SchemaTableName("default", topicName),
+                        createDescription(topicName, "default", topicName,
+                                createFieldGroup("raw", ImmutableList.of(
+                                        createOneFieldDescription("id", BigintType.BIGINT, "0", "LONG")))))
+                .build();
+
+        QueryRunner queryRunner = KafkaQueryRunner.builder(testingKafka)
                 .setTables(TpchTable.getTables())
+                .setExtraTopicDescription(ImmutableMap.<SchemaTableName, KafkaTopicDescription>builder()
+                        .putAll(extraTopicDescriptions)
+                        .build())
                 .build();
+
+        testingKafka.createTopics(topicName);
+        return queryRunner;
+    }
+
+    @Test
+    public void testWhereClause()
+    {
+        ByteBuffer buf = ByteBuffer.allocate(8);
+        buf.putLong(0, 1);
+
+        insertData(buf.array());
+
+        assertQuery("SELECT id FROM default." + topicName + " WHERE id = 1", "VALUES (1)");
```

Comment:
engine could optimize this query, knowing `id` is 1, it doesn't need to load it twice, only once for the check

run a second query with `< 2` condition

Reply:
that's a smart engine

Reply:
i don't think it is --  but could be/become